tracer/src/Datadog.Trace/Logging/TracerFlare/TracerFlareManager.cs

```diff
@@ -341,10 +348,32 @@ static bool TryGetArg(JObject argsObject, string name, string configPath, [NotNu
         }
     }
 
-    private static bool IsEnableDebugConfig(RemoteConfigurationPath remoteConfigPath)
+    private static bool IsEnableDebugConfig(RemoteConfiguration remoteConfig)
     {
-        return remoteConfigPath.Id.Equals("flare-log-level.debug", StringComparison.Ordinal)
-            || remoteConfigPath.Id.Equals("flare-log-level.trace", StringComparison.Ordinal);
+        try
+        {
+            var remoteConfigPath = remoteConfig.Path;
+
+            if (remoteConfigPath.Id.Equals("flare-log-level.debug", StringComparison.Ordinal)
+                || remoteConfigPath.Id.Equals("flare-log-level.trace", StringComparison.Ordinal))
+            {
+                return true;
+            }
+
+            var json = JObject.Parse(EncodingHelpers.Utf8NoBom.GetString(remoteConfig.Contents));
+
+            var logLevel = json["config"]?["log_level"]?.Value<string>();
+
+            return logLevel is not null
+                && (logLevel.Equals("debug", StringComparison.OrdinalIgnoreCase)
+                    || logLevel.Equals("trace", StringComparison.OrdinalIgnoreCase));
+        }
+        catch (Exception ex)
+        {
+            Log.Debug(ex, "Invalid configuration provided for tracer flare");
```

Comment:
I feel like this should _probably_ be an error log? We would probably want to know about it? 🤔

Reply:
I agree but here we made it a debug log: https://github.com/DataDog/dd-trace-dotnet/blob/master/tracer/src/Datadog.Trace/Logging/TracerFlare/TracerFlareManager.cs#L257

So I don't know 🤷